src/ADAL.PCL/Http/AdalHttpClient.cs

```diff
@@ -101,6 +100,13 @@ public AdalHttpClient(string uri, CallState callState)
                     if (ex.WebResponse != null)
                     {
                         TokenResponse tokenResponse = TokenResponse.CreateFromErrorResponse(ex.WebResponse);
+
+                        // Check for claims challenge response
+                        if (!string.IsNullOrEmpty(tokenResponse.Claims))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)